### PR TITLE
Online editor: Cleanup TS setup

### PR DESCRIPTION
**Pull request by hunger**
Reformat the code and config files, set up linting, update typescript version, set up TS to find all the modules, enable `strict` mode in TS, fix all issues reported by tsc in strict mode, have a npm script to run `tsc --noemit` as recommended by vite.

I do not expect any of this to cause problems in CI as these changes.

### CODE REVIEW

**Comment by tronical** [comment]
Since we recently had the topic with CMake files, I'm curious: What is the formatter used for the files, and how do we make sure that we all end up formatting the file the same way?

File: tools/online_editor/package.json [hunk 1]
```diff
@@ -10,7 +10,6 @@
   "keywords": [],
   "author": "",
   "dependencies": {
-    "@types/jquery": "^3.5.4",
```
Comment (ogoffart) [inline tools/online_editor/package.json:13]: yay :+1:

File: tools/online_editor/package.json [hunk 2]
```diff
@@ -4,6 +4,7 @@
   "description": "",
   "main": "index.js",
   "scripts": {
+    "syntax_check": "tsc --noemit",
```
Comment (ogoffart) [inline tools/online_editor/package.json:7]: can we run it in CI?
alternatively , can we configure typescript compiler to actually error out in the CI?

Comment (ogoffart) [reply tools/online_editor/package.json:7]: actually, it seems another commit do that, so that should be fine

Comment (hunger) [reply tools/online_editor/package.json:7]: No, it now the Ci errors out because I somehow broke the editor worker for Monaco.

**Comment by hunger** [comment]
@tronical: I use prettier for basically everything not handled by rustfmt :-) Markdown, json, html, js, ts, ... it re-formats pretty much everything:-)

**Comment by tronical** [comment]
> @tronical: I use prettier for basically everything not handled by rustfmt :-) Markdown, json, html, js, ts, ... it re-formats pretty much everything:-)

Ok, and (I just tried) when I format these files with VS code I get different output for example for `tsconfig.json`.

**Comment by hunger** [comment]
Every tool does something different wrt. formatting and that VScode has several dozen plugins that format code in the web realm does not help either.

I am sure you can configure your VSCode to use prettier though: That seems very popular.

Even if not: Use pre-commit: It will format using prettier for you on git commit :-)

**Comment by hunger** [comment]
CI is now doing the compile/lint steps. They are both <5s on my laptop, so I just pushed them into the existing build workflow.

**Comment by ogoffart** [comment]
We should have the formating config file pushed to the repo, and the CI to enforce the formatting
Otherwise, don't reformat.
This is just my opinion.

File: tools/online_editor/package.json (hunk 2 above)
Comment (hunger) [reply tools/online_editor/package.json]: added that and the linter run to the CI in the meantime.

**Pull request merged**